control-plane-operator/controllers/hostedcontrolplane/kas/kms/ibmcloud.go

```diff
@@ -50,7 +50,7 @@ type ibmCloudKMSProvider struct {
 }
 
 func NewIBMCloudKMSProvider(ibmCloud *hyperv1.IBMCloudKMSSpec, kmsImage string) (*ibmCloudKMSProvider, error) {
-	if ibmCloud == nil || len(ibmCloud.KeyList) == 0 || len(ibmCloud.Region) == 0 || len(kmsImage) == 0 {
```

Comment:
I believe this has been broken since https://github.com/openshift/hypershift/pull/3183. The empty struct at https://github.com/openshift/hypershift/blob/43412b9b44ce8e43da2ede2f09852258dde5a9c8/control-plane-operator/controllers/hostedcontrolplane/kas/kms.go#L25 means that `kmsImage` is always empty here.

Reply:
nice catch!